### ISSUE TITLE
EfficientNet requires TF2.2

### ISSUE THREAD

**Issue by fotiDim** [doc-ai/tensorio#39]
I am trying to run inference on an EfficientNet [model](http://download.tensorflow.org/models/object_detection/tf2/20200711/efficientdet_d0_coco17_tpu-32.tar.gz) from TF2 model [zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md). It fails with the following errors. I think a TF2.2 binary is needed from what I read [here](https://tfhub.dev/tensorflow/efficientnet/b0/classification/1). Is there any change we can have a TF2.2 release?
```
2020-08-18 15:27:18.628392: W tensorflow/core/framework/op_kernel.cc:1622] OP_REQUIRES failed at save_restore_v2_ops.cc:184 : Data loss: Unable to read file (/private/var/containers/Bundle/Application/12DD2E17-6F5F-4605-8A55-5483B261AEFB/TF114.app/models/efficientdet_d1_coco17_tpu-32.tiobundle/predict/variables/variables.index). Perhaps the file is corrupt or was produced by a newer version of TensorFlow with format changes (failed to seek to header entry): corrupted compressed block contents
2020-08-18 15:27:18.637753: I tensorflow/cc/saved_model/loader.cc:311] SavedModel load for tags { serve }; Status: fail. Took 927635 microseconds.
2020-08-18 15:27:18.637810+0200 TF114[3559:329525] Unable to load saved model, status: Data loss: [_Derived_]{{function_node __inference__traced_restore_144300}} {{function_node __inference__traced_restore_144300}} Unable to read file (/private/var/containers/Bundle/Application/12DD2E17-6F5F-4605-8A55-5483B261AEFB/TF114.app/models/efficientdet_d1_coco17_tpu-32.tiobundle/predict/variables/variables.index). Perhaps the file is corrupt or was produced by a newer version of TensorFlow with format changes (failed to seek to header entry): corrupted compressed block contents
	 [[{{node RestoreV2}}]]
	 [[StatefulPartitionedCall_2]]
2020-08-18 15:27:18.637950+0200 TF114[3559:329525] There was a problem loading the model from run:error:, error: Error Domain=ai.doc.tensorio Code=104 "Saved model could not be loaded from the model bundle" UserInfo={NSLocalizedDescription=Saved model could not be loaded from the model bundle}
```

**Comment by phildow** [comment]
Hi @fotiDim ,

Have you tried building that model on the python side in TF 2.0? Just downgrade your python tensorflow dependency and see if it works.

Unfortunately it is unlikely we will ever see any support for greater than TF 2.0 on iOS. The Google team that manages the public TensorFlow repo, which our works is based on, removed all of the files for building TensorFlow for iOS in TF 2.1+ That code had been deprecated for quite some time, and they finally got rid of it.

However, many models built in TF 2.2 can be built in TF 2.0 on the python side, in which case it might still work.

You may still see another, separate error, for that particular model architecture, which occurs when the build of tensorflow doesn't include the required kernel for a particular op. That's a separate issue, and as long as a kernel is available for it, they're usually simple enough to enable and then you'd have to rebuild TensorFlow.

**Comment by fotiDim** [comment]
I haven't tried rebuilding the model with TF2.0. I have only tried inference with TF2.0 with Python on Linux and this also failed.
I would expect building the model with TF2.0 would fail as it clearly states `This model requires TensorFlow 2.2 or newer, because it contains a FusedBatchNormV3 op with attr exponential_avg_factor.`

I see. That's a bummer. I am closing this issue as there is nothing to be done apart from the potential workaround you mention.

**Comment by phildow** [comment]
Try rebuilding the model with TF 2.2 and not TF 2.0. We've tested TF 2.2 models with our on-device 2.0 build and they've worked.

Also have a look at our [TF 2.2 CIFAR10 Saved Model Example](https://github.com/doc-ai/tensorio/blob/master/examples/notebooks/tf_2_2/keras_cifar10_mobilenet_save_model/keras_cifar10_mobilenet_save_model.ipynb) and specifically this line [27]:

```python
tf.config.optimizer.set_experimental_options({
    'remapping': False
})
```

This setting disables those fused op kernels, which we needed to do in order to run even basic MobileNet models from TF 2.2 with our on-device build. Those fused kernels are optimizations which would be nice to have but which we couldn't get support for.
